src/libcore/macros.rs

```diff
@@ -32,8 +32,19 @@ macro_rules! panic {
 
 /// Ensure that a boolean expression is `true` at runtime.
 ///
-/// This will invoke the `panic!` macro if the provided expression cannot be
-/// evaluated to `true` at runtime.
+/// This will ensure the termination of the program if the provided expression
+/// cannot be evaluated to `true` at runtime by means of an unrecoverable error
+/// (not necessarily a `panic!`, can also be an `abort`).
```

Comment:
So, here's my one concern with this wording: it _will_ always panic, the key is that panic can be either unwinding or abort. I would just stick to saying that this panics.